src/main/scala/com/wanari/tutelar/core/impl/ConfigServiceImpl.scala

```diff
@@ -314,12 +319,38 @@ class ConfigServiceImpl() extends ConfigService {
 
   private def readOauth2Config(name: String): OAuth2Config = {
     Try {
-      val config = conf.getConfig(s"oauth2.$name")
+      val config   = conf.getConfig(s"oauth2.$name")
+      val authType = config.getString("auth")
+      val auth = authType match {
+        case "basic" =>
+          ServiceAuthDirectives.BasicAuthConfig(
+            config.getString("basic.username"),
+            readFromFileOrConf(config, "basic.password")
+          )
+        case "escher" =>
+          ServiceAuthDirectives.EscherAuthConfig(
+            config.getString("escher.trustedServices").split(",").toList
+          )
+        case "jwt" =>
+          val configName = config.getString("jwt.configName")
+          JwtAuthConfig(getJwtConfigByName(configName))
+        case "custom_header" =>
+          ServiceAuthDirectives.CustomHeaderAuthConfig(
+            config.getString("customHeader.headername"),
+            readFromFileOrConf(config, "customHeader.secret")
+          )
+        case "blocked" =>
+          AccessBlocked
+        case t => throw new IllegalArgumentException(s"$t unknown service auth type in $name.")
+      }
+
```

Comment:
This whole part is already implemented at `getServiceAuthConfig`, i think you should use that.

Reply:
or.. why need this field to this config?